derive(Debug) does not compile

Hi,

The following does not compile:
```rust
#[ouroboros::self_referencing]
#[derive(Debug)]
pub struct Struct<'a> {
    bytes: Vec<u8>,
    #[borrows(bytes)]
    slice: &'a [u8],
}
```

```
error: bad syntax
 --> src/lib.rs:4:1
  |
4 | #[derive(Debug)]
  | ^^^^^^^^^^^^^^^^

error: could not compile `ouroboros-private` (lib) due to previous error
```

Neither does this:
```rust
#[derive(Debug)]
#[ouroboros::self_referencing]
pub struct Struct<'a> {
    bytes: Vec<u8>,
    #[borrows(bytes)]
    slice: &'a [u8],
}
```

```
    Checking ouroboros-private v0.1.0 (/home/maxence/bug/ouroboros-private)
error[E0609]: no field `bytes` on type `&Struct<'a>`
 --> src/lib.rs:4:5
  |
4 |     bytes: Vec<u8>,
  |     ^^^^^ unknown field

error[E0609]: no field `slice` on type `&Struct<'a>`
 --> src/lib.rs:6:5
  |
6 |     slice: &'a [u8],
  |     ^^^^^ unknown field

For more information about this error, try `rustc --explain E0609`.
error: could not compile `ouroboros-private` (lib) due to 2 previous errors
```

Im using `rustc 1.72.0-nightly (1c53407e8 2023-05-28)`

--- a/ouroboros_macro/src/parse.rs
+++ b/ouroboros_macro/src/parse.rs
@@ -1,6 +1,6 @@
 use proc_macro2::{Delimiter, Span, TokenTree};
 use quote::format_ident;
-use syn::{spanned::Spanned, Attribute, Error, Fields, GenericParam, ItemStruct, Meta};
+use syn::{spanned::Spanned, Attribute, Error, Fields, GenericParam, ItemStruct, Meta, MacroDelimiter};
 
 use crate::{
     covariance_detection::type_is_covariant_over_this_lifetime,
@@ -116,11 +116,14 @@ fn parse_derive_token(token: &TokenTree) -> Result<Option<Derive>, Error> {
 
 fn parse_derive_attribute(attr: &Attribute) -> Result<Vec<Derive>, Error> {
     let body = match &attr.meta {
-        Meta::List(ml) => &ml.tokens,
+        Meta::List(ml) => ml,
         _ => unreachable!(),
     };
+    if !matches!(body.delimiter, MacroDelimiter::Paren(_)){
+        return Err(Error::new(attr.span(), format!("malformed derive input, derive attributes are of the form `#[derive({})]`",body.tokens.to_string())));
+    }
     let mut derives = Vec::new();
-    for token in body.clone().into_iter() {
+    for token in body.tokens.clone().into_iter() {
         if let Some(derive) = parse_derive_token(&token)? {
             derives.push(derive);
         }

--- a/ouroboros_macro/src/parse.rs
+++ b/ouroboros_macro/src/parse.rs
@@ -119,20 +119,13 @@ fn parse_derive_attribute(attr: &Attribute) -> Result<Vec<Derive>, Error> {
         Meta::List(ml) => &ml.tokens,
         _ => unreachable!(),
     };
-    if let Some(TokenTree::Group(body)) = body.clone().into_iter().next() {
-        if body.delimiter() != Delimiter::Parenthesis {
-            panic!("TODO: nice error, bad define syntax")
+    let mut derives = Vec::new();
+    for token in body.clone().into_iter() {
+        if let Some(derive) = parse_derive_token(&token)? {
+            derives.push(derive);
         }
-        let mut derives = Vec::new();
-        for token in body.stream().into_iter() {
-            if let Some(derive) = parse_derive_token(&token)? {
-                derives.push(derive);
-            }
-        }
-        Ok(derives)
-    } else {
-        Err(Error::new(attr.span(), "bad syntax"))
     }
+    Ok(derives)
 }
 
 pub fn parse_struct(def: &ItemStruct) -> Result<StructInfo, Error> {

